org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/StringUtils.java

```diff
@@ -176,6 +176,43 @@ public static String lTrim(String value) {
 		return value.substring(i, len);
 	}
 
-	
+	/**
+	 * Given a string that is only whitespace,
+	 * this will return the amount of newline characters.
+	 * 
+	 * If the newLineStopCount > newLineCounter, then the value of
+	 * newLineStopCount is always returned.
+	 * @param text
+	 * @param isWhitespace
+	 * @param delimiter
+	 * @return
+	 */
+	public static int getNumberOfNewLines(String text, boolean isWhitespace, String delimiter) {
+		if(!isWhitespace){
+			return 0;
+		}
+		int newLineStopCount = 3;

```

Comment:
Number seems too arbitrary. Might has well use PRESERVE_NEWLINES as the preference setting that limit, instead of using a boolean